### PR TITLE
cleaning up azure_rm_deployment samples & return value doc

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Just some naming conventions, etc.
Added **name** parameter in every sample
Removed some unnecessary things like **subscription_id**
Fixed & extended return value documentation

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
azure_rm_deployment

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/112336/87/tests

File: lib/ansible/modules/cloud/azure/azure_rm_deployment.py [hunk 1]
```diff
@@ -335,30 +311,82 @@
...
+        id:
+            description: The Azure ID of the deployment
+            type: str
+            returned: always
+            sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Resources/deployments/myD
+                     eployment"
+        instances:
+            description: Provides the public IP addresses for each VM instance.
+            type: list
+            returned: always
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_deployment.py:332]: return always? what if my arm template doesn't contain VM?

Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_deployment.py:332]: well, field is always returned, it may just be empty []

Comment (yungezz) [reply lib/ansible/modules/cloud/azure/azure_rm_deployment.py:332]: it's confusing, such as public  ip marked as returned always also.

Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_deployment.py:332]: yes, the lists are always returned, just may be empty,
maybe we could have:
**returned: always but list may be empty**

or just
**returned when applicable**

**Review by yungezz — APPROVED** [review]

File: lib/ansible/modules/cloud/azure/azure_rm_deployment.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_deployment.py:332]: I actually think "instances" return value should be made obsolete.
it works only with vms.
if somebody uses ARM templates to deploy anything else than VM it's pretty much useless.
we could think of more universal approach.

Usually **outputs** should be sufficient to use created resources. For instance for Windows VM, outputs look like:
```json
            "outputs": {
                "hostname": {
                    "type": "String",
                    "value": "zimsdplxxxx.eastus2.cloudapp.azure.com"
                }
            }
```

So it has sufficient information to use the VM.
Outputs are designed by ARM template authors so it's up to them what's important for the user.

**Pull request merged**